### DISCUSSION TITLE
Is there a similar app for YouTube music?

### DISCUSSION THREAD

**Discussion by lantai5** [TeamNewPipe/NewPipe#11042] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my question isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the question will be dismissed otherwise.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### What is/are your question(s)?

Youtube music, even though I am a paying user, does not allow me to download/cache multiple songs at once manually... They only way is allow auto downloading, because google know us better than we do? Anyway, maybe someone can make a way to allow multi track caching manually.

**Comment by fillwithjoy1** [comment]
I don't know about the team here, but I'll hook you up with some options outside of NewPipe.

1. [InnerTune](https://github.com/z-huang/InnerTune)
2. [YouTube Vanced](https://github.com/TeamVanced/VancedManager) (brings back the old design if you liked it)
3. [YouTube ReVanced](https://github.com/ReVanced/revanced-manager) (you need to patch your own youtube music app. If you want rootless install, patch the google play services to microG and download the revanced version of [microG](https://github.com/ReVanced/GmsCore/releases/tag/v0.3.1.4.240913)

**Comment by opusforlife2** [comment]
I know ViMusic (on Fdroid) does cache tracks. But I'm not sure if that's done for multiple tracks.

**Reply by ikanakova** [reply]
ViMusic is dead (no further development and no bug fixes), but there are other apps like RiMusic, InnerTune, SimpMusic, SpMp...

**Reply by brnwlshubh** [reply]
Also you can use ViTune

**Reply by opusforlife2** [reply]
I didn't realise RiMusic was a continued fork of ViMusic. That's awesome. It's a great app.

**Comment by ikanakova** [comment]
Definitely [RiMusic](https://github.com/fast4x/rimusic/), which allow multi track caching manually.
